### PR TITLE
useLiteralContent data validation

### LINKED ISSUE #5456: StandardJSON crashes on useLiteralContent not bool
Found by solfuzzer.

Problematic line here: https://github.com/ethereum/solidity/blob/develop/libsolidity/interface/StandardCompiler.cpp#L409

Need to add tests for `useLiteralContent` not present.

### PR DESCRIPTION
**Pull request by maxsam4**
### Description
- Added datatype check for `settings.metadata.useLiteralContent`
- Fixed #5456 

### Checklist
- [X] Code compiles correctly
- [X] All tests are passing
- [X] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `test_check_style` for commit `5a7cf08db3027d75081d2e698dd51b807bef5abb`:
```
Error: Format error for if/for:
 libsolidity/interface/StandardCompiler.cpp:581: if(settings.isMember("metadata")) {
 libsolidity/interface/StandardCompiler.cpp:581: if(settings.isMember("metadata")) {

```
Please check that your changes are working as intended.

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -298,6 +298,8 @@ boost::optional<Json::Value> checkOptimizerKeys(Json::Value const& _input)
 
 boost::optional<Json::Value> checkMetadataKeys(Json::Value const& _input)
 {
+	if (_input.isObject() && _input.isMember("useLiteralContent") && !_input["useLiteralContent"].isBool())
+		return formatFatalError("JSONError", "\"settings.metadata.useLiteralContent\" must be a bool");
```
Comment (leonardoalt) [inline libsolidity/interface/StandardCompiler.cpp]:
```suggestion
		return formatFatalError("JSONError", "\"settings.metadata.useLiteralContent\" must be Boolean");
```

File: libsolidity/interface/StandardCompiler.cpp [hunk 2]
```diff
@@ -576,12 +578,13 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 	}
 	m_compilerStack.setLibraries(libraries);
 
-	Json::Value metadataSettings = settings.get("metadata", Json::Value());
-
-	if (auto result = checkMetadataKeys(metadataSettings))
-		return *result;
-
-	m_compilerStack.useMetadataLiteralSources(metadataSettings.get("useLiteralContent", Json::Value(false)).asBool());
+	if (settings.isMember("metadata"))
```
Comment (chriseth) [inline libsolidity/interface/StandardCompiler.cpp]: The style of the other lines suggests that this check is not needed, or is it?

File: test/cmdlineTests/standard_missing_type_useLiteralContent/input.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (chriseth) [inline test/cmdlineTests/standard_missing_type_useLiteralContent/input.json]: Shouldn't the file be named "standard_missing_key_..."?

File: libsolidity/interface/StandardCompiler.cpp (hunk 2 above)
Comment (leonardoalt) [reply libsolidity/interface/StandardCompiler.cpp]: I had the same question.

**Review by chriseth — CHANGES_REQUESTED** [review]
Nice, that's almost it already!

**Comment by leonardoalt** [comment]
Thanks @maxsam4 !

@chriseth Didn't we have a PR that validated every field?

File: libsolidity/interface/StandardCompiler.cpp (hunk 2 above)
Comment (maxsam4) [reply libsolidity/interface/StandardCompiler.cpp]: Optional objects are using this check. For example, `if (settings.isMember("optimizer"))` is present on L514 so I added this for consistency. That being said, it was working earlier as well when `metadata` was absent so we probably don't need the check.

File: test/cmdlineTests/standard_missing_type_useLiteralContent/input.json (hunk 3 above)
Comment (maxsam4) [reply test/cmdlineTests/standard_missing_type_useLiteralContent/input.json]: Yep, changed the name.

File: libsolidity/interface/StandardCompiler.cpp (hunk 2 above)
Comment (maxsam4) [reply libsolidity/interface/StandardCompiler.cpp]: Everything is working fine without this check as well. Let me know if you want to keep the check or remove it. The check can be a minor optimization if `settings.metadata` is mostly missing. Also makes it more explicit that `settings.optimizer` is optional.

Comment (chriseth) [reply libsolidity/interface/StandardCompiler.cpp]: Please remove the check

Comment (maxsam4) [reply libsolidity/interface/StandardCompiler.cpp]: Removed.

**Comment by leonardoalt** [comment]
Looks good! Could you please squash the commits?

**Comment by maxsam4** [comment]
> Looks good! Could you please squash the commits?

Thanks. 
Just force pushed a squashed commit with the default squash message from git.

**Review by chriseth — APPROVED** [review]
Thanks a lot!

**Comment by maxsam4** [comment]
It was a pleasure working on this. I'll definitely pick up more tasks in my free time :)

**Pull request merged**